yaml-tests/src/test/resources/struct-type-nullability-variants.yamsql

```diff
@@ -0,0 +1,123 @@
+#
+# struct-type-nullability-variants.yamsql
+#
+# This source file is part of the FoundationDB open source project
+#
+# Copyright 2021-2024 Apple Inc. and the FoundationDB project authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Tests for struct types with same name but different nullability
+# This validates that a struct type like "Address" can exist in both:
+# - Nullable form (for table columns)  
+# - Non-nullable form (for array elements)
+---
+options:
+    supported_version: !current_version
+---
+schema_template:
+    create type as struct Address(street string, city string, zipcode integer)
+    
+    create table users(
+        id integer, 
+        name string, 
+        home_address Address,
+        primary key(id)
+    )
+    
+    create table locations(
+        id integer,
+        name string,
+        addresses Address array,
+        primary key(id)
+    )
+    
+    create table businesses(
+        id integer,
+        name string,
+        headquarters Address,
+        branch_offices Address array,
```

Comment:
If it I take it, the reason this covers the new logic is that the `Address array` will be an array of non-nullable `Address` structs, whereas the `headquarters Address` will be a nullable type. It might make sense to add an explicit table here where there is both an `Address` column and an `Address NOT NULL` column.